File: MATH2232 Numerical Methods Lab/Numerical Analysis Program/Secant.cpp
#include<bits/stdc++.h>
using namespace std;
#define EPSILON 0.001


double func(double x){
    return x*x*x - x*x + 2;
}

void secant(double x0, double x1){
    double dx1 = (func(x1) - func(x0)) / (x1 - x0);
    double x2 = x1 - (func(x1) / dx1);

    while (func(x2) >= EPSILON)
    {
       x2 = x1 - (func(x1) / dx1);

       x0 = x1;
       x1 = x2;
            
    }
        
    cout << "The value of Root is : " << x2 <<endl;

}



int main()
{
    double x0 = -10,x1 = 10;
    secant(x0,x1);

    return 0;
}

File: MATH2232 Numerical Methods Lab/RU/Numerical-Methods-Lab-Final-main akif/Secant.java
package Akif;

public class Secant {

    // A function just holding the equation
    public static double func(double x) {
        return Equation.solve(x);
    }

    public static void Secant(double x0, double x1, double EPSILON) {

        int iteration = 0;

        if (func(x0) * func(x1) > 0) {
            System.out.println("Wrong Guess");
            return;
        }


        while (Math.abs(x1 - x0) >= EPSILON) {
            double x2 = ((x0 * func(x1)) - (x1 * func(x0))) / (func(x1) - func(x0));
            x0 = x1;
            x1 = x2;
            System.out.println("Iteration :" + (++iteration) + ", Root = " + x2);
        }
    }


}

